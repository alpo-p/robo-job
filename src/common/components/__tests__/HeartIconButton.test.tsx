import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import HeartIconButton from '../HeartIconButton'

describe('HeartIconButton', () => {
  const mockPress = jest.fn()

  it('should match snapshot', () => {
    const tree = renderer.create(
      <HeartIconButton isPressed={false} onPress={mockPress} />,
    )
    expect(tree).toMatchSnapshot()
  })
  it('should register a press', () => {
    const { container } = render(
      <HeartIconButton isPressed={false} onPress={mockPress} />,
    )
    fireEvent.press(container)
    expect(mockPress).toHaveBeenCalledTimes(1)
  })
})
