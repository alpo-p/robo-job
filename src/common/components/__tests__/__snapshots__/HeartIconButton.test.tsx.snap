// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeartIconButton should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
